kroxylicious-kubernetes-api/src/main/java/io/kroxylicious/kubernetes/api/common/tls/CipherSuites.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.api.common.tls;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import com.fasterxml.jackson.annotation.JsonSetter;
+
+/**
+ * TLS CipherSuite controls.
+ */
+@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)
+@com.fasterxml.jackson.annotation.JsonPropertyOrder({ "allowed", "denied" })
+@com.fasterxml.jackson.databind.annotation.JsonDeserialize(using = com.fasterxml.jackson.databind.JsonDeserializer.None.class)
+@javax.annotation.processing.Generated("io.fabric8.java.generator.CRGeneratorRunner")
+@lombok.ToString()
+@lombok.EqualsAndHashCode()
+@io.sundr.builder.annotations.Buildable(editableEnabled = false, validationEnabled = false, generateBuilderPackage = false, builderPackage = "io.fabric8.kubernetes.api.builder", refs = {
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ObjectMeta.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ObjectReference.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.LabelSelector.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.Container.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.EnvVar.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ContainerPort.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.Volume.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.VolumeMount.class)
+})
+public class CipherSuites implements io.fabric8.kubernetes.api.builder.Editable<CipherSuitesBuilder>, io.fabric8.kubernetes.api.model.KubernetesResource {
+
+    @Override
+    public CipherSuitesBuilder edit() {
+        return new CipherSuitesBuilder(this);
+    }
+
+    /**
+     * The allowed cipher suites.
+     * If not specified a default list of ciphers, which depends on the runtime and enabled protocols, will be used.
+     */
+    @JsonProperty("allowed")
```

Comment:
I wondered about this when I did the work for `KafkaService.spec.tls`... should it be `allowed` or `allow`? I think the active tense of `allow` actually fits better with the notion that a resource is declarative.

Reply:
Done.